### PR TITLE
VDS-1374: Set limits of the file to upload

### PR DESCRIPTION
**Pull request by ilyawzrd**
Download artifact URL: https://vc3prerelease.blob.core.windows.net/packages/VirtoCommerce.CustomerExportImport_1.0.0-pr-12-99285171.zip

### CODE REVIEW

**Review by trueboroda — APPROVED** [review]

File: src/VirtoCommerce.CustomerExportImportModule.Data/Services/CsvCustomerImportHelper.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+using System;
+using System.Linq;
+using CsvHelper.Configuration.Attributes;
+using VirtoCommerce.CustomerExportImportModule.Data.Models;
+
+namespace VirtoCommerce.CustomerExportImportModule.Data.Services
+{
+    public static class CsvCustomerImportHelper
+    {
+        public static string[] GetImportPriceRequiredColumns()
```
Comment (pushnitsa) [inline src/VirtoCommerce.CustomerExportImportModule.Data/Services/CsvCustomerImportHelper.cs:10]: GetImportCustomerRequiredColumns

**Review by pushnitsa — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.CustomerExportImportModule.Data/Services/CsvCustomerDataValidator.cs [hunk 2]
```diff
@@ -100,5 +104,80 @@ private void ValidateLineLimit(Stream stream, Configuration csvConfiguration, Li
...
+            var csvReader = new CsvReader(streamReader, csvConfiguration);
+
+            csvReader.Read();
+            csvReader.ReadHeader();
+
+            var existedColumns = csvReader.Context.HeaderRecord;
+
+            var requiredColumns = CsvCustomerImportHelper.GetImportPriceRequiredColumns();
+
+            var missedColumns = requiredColumns.Except(existedColumns).ToArray();
```
Comment (pushnitsa) [inline src/VirtoCommerce.CustomerExportImportModule.Data/Services/CsvCustomerDataValidator.cs:134]: comparing should be case insensitive. Please add the second parameter,  **StringComparer.InvariantCultureIgnoreCase**

**Review by pushnitsa — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.CustomerExportImportModule.Data/Services/CsvCustomerImportHelper.cs (hunk 1 above)
Comment (ilyawzrd) [reply src/VirtoCommerce.CustomerExportImportModule.Data/Services/CsvCustomerImportHelper.cs]: fixed

File: src/VirtoCommerce.CustomerExportImportModule.Data/Services/CsvCustomerDataValidator.cs (hunk 2 above)
Comment (ilyawzrd) [reply src/VirtoCommerce.CustomerExportImportModule.Data/Services/CsvCustomerDataValidator.cs]: done

**Review by pushnitsa — APPROVED** [review]

**Pull request merged**